department-of-veterans-affairs/vets-website
# 74398 Use preferred name from demographics

## Summary
Preferred name is now part of API response of `/v0/user`. This PR adds preferred name to the state and changes the MHV landing page to use it. This eliminates and extra API call when loading the landing page and removes the need to wait for the API call to finish before rendering the page.

## Related issue(s)
https://app.zenhub.com/workspaces/mhv-on-vagov-landing-page-62619a987d74510018ecc546/issues/gh/department-of-veterans-affairs/va.gov-team/74398

## Testing done
- Updated unit tests
Manual testing:
1. Checkout the latest branch for the `vets-api-mockdata` repo
1. Checkout the latest branch for the `vets-api` repo
1. Checkout **this** branch for the `vets-website` repo
1. Start the `vets-api` locally
1. Start the `vets-website` locally
1. Access the [local mock login](http://localhost:3001/sign-in/mocked-auth)
1. Login as one of the following IDME users that have different preferred names in the mock data:
 - `vets.gov.user+0@gmail.com`
 - `vets.gov.user+1@gmail.com`
 - `vets.gov.user+10@gmail.com` - this one has preferred name = null
8. Navigate to the MHV Landing Page and verify the preferred name is shown correctly

## Screenshots
N/A. There are no changes to the UI. For reference, a screenshot of how the prefer name shows up in the landing page is below.
![localhost_3001_my-health_ (5)](https://github.com/department-of-veterans-affairs/vets-website/assets/63804190/38db22ae-8f52-4ba8-984e-be2bbc5e44f7)

## What areas of the site does it impact?
- Adds demographics to the state
- MHV landing page

## Acceptance criteria

### Quality Assurance & Testing


### Error Handling


### Authentication


### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:


## Requested Feedback
- Feedback on location of demographics data in the state
